### DISCUSSION TITLE
Is it possible to set filetype & syntax for `:MetalsToggleLogs`?

### DISCUSSION THREAD

**Discussion by jsatk** [scalameta/nvim-metals#207] [Q&A] [answered]
Basically the title. If you run `:MetalsToggleLogs` it tails the file in a split and has no filetype or syntax. If you open it via `:vs .metals/metals.log` it has `filetype=log` and `syntax=log`. Is there a way to get syntax highlight for `log` via `:MetalsToggleLog`? Or is that just not do-able because of `tail`? If it's the latter I wonder if there's a way to use `bat` if the user has it to produce syntax highlighted logs. 🤔

**Comment by ckipp01** [comment, marked as answer]
This is totally doable. For example I just tested this with:

```
:vsp +set\ ft=log term://tail -f .metals/metals.log
```

Which will open the split, set the ft, and then enter term and tail the log. It should be as simple as changing this line to reflect it https://github.com/scalameta/nvim-metals/blob/377d7f9e325bc488ac750d490ef48fd55f33bda9/lua/metals.lua#L172

**Reply by jsatk** [reply]
Well hot damn. Thank you.

**Reply by tkolleh** [reply]
How can the split be set to `s` when [using telescope](https://github.com/tkolleh/dotfiles/blob/9ecb689a3a12e912d89c42bf553fe53800c91c2b/private_dot_config/lvim/lua/user/metals.lua#L70)?

**Reply by ckipp01** [reply]
Hey @tkolleh I'm actually not sure. The ability to control the split to vsp or sp was recently added and works when doing `:MetalsToggleLogs sp` for example, but I haven't yet look into how to pass an arg with telescope look this. Feel free to create a feature request for this.
